routini/tests/proxy/upstream_peer.rs

```diff
@@ -0,0 +1,138 @@
+use reqwest::StatusCode;
+use routini::{
+    load_balancing::strategy::Adaptive,
+    server_builder::{Route, RouteConfig},
+};
+
+use super::helpers::TestApp;
+
+/// Tests that a request to a non-existent route returns BAD_REQUEST
```

Comment:
It really should just return a 404.  The load balancer should be transparent from the client's point of view.